src/core/stream/period/period_stream.ts

```diff
@@ -157,29 +162,37 @@ export default function PeriodStream({
       const newStream$ = clock$.pipe(
         take(1),
         mergeMap((tick) => {
+          const { audioTrackSwitchingMode } = options;
           const qSourceBuffer = createOrReuseQueuedSourceBuffer(sourceBuffersStore,
                                                                 bufferType,
                                                                 adaptation,
                                                                 options);
           const strategy = getAdaptationSwitchStrategy(qSourceBuffer,
                                                        period,
                                                        adaptation,
-                                                       tick);
+                                                       tick,
+                                                       audioTrackSwitchingMode);
           if (strategy.type === "needs-reload") {
             return observableOf(EVENTS.needsMediaSourceReload(period, tick));
           }
 
           const cleanBuffer$ = strategy.type === "clean-buffer" ?
             observableConcat(...strategy.value.map(({ start, end }) =>
                                qSourceBuffer.removeBuffer(start, end))
-                            ).pipe(ignoreElements()) :
-            EMPTY;
+                            ).pipe(last())
+                            : EMPTY;
 
           const bufferGarbageCollector$ = garbageCollectors.get(qSourceBuffer);
-          const adaptationStream$ = createAdaptationStream(adaptation, qSourceBuffer);
+          const adaptationStream$ = createAdaptationStream(adaptation,
+                                                           qSourceBuffer);
```

Comment:
Still unneeded line break :p

Reply:
Thanks, Haha, I m not enough use to the new code formatter